### PR TITLE
Repair nin init

### LINKED ISSUE #278: nin init does not work
It broke when we moved to a node-based system.

### PR DESCRIPTION
**Pull request by sigvef**
This fixes #278.

### CODE REVIEW

File: nin/backend/generate/generate.js [hunk 1]
```diff
@@ -19,10 +19,14 @@ var generate = function(projectRoot, type, name) {
...
 
       graph.transform(projectRoot, function(g) {
-        g.push({
+        const object = {
           id: name,
-          type: camelizedName
-        });
+          type: camelizedName,
+        };
+        for(const key in options) {
```
Comment (aleksanb) [inline nin/backend/generate/generate.js]: You can replace this with the new (and sexy) `Object.assign(object, options)` 🍾

Comment (stianjensen) [inline nin/backend/generate/generate.js]: Can't you also use spread syntax inside the object literal?

Comment (aleksanb) [inline nin/backend/generate/generate.js]: Yeah, that might work as well!
What a time to be alive!

**Comment by sigvef** [comment]
> Can't you also use spread syntax inside the object literal?

That's what I was trying to pull off on Friday, to no avail. Can you show how that would work? If not I'll go with Object.assign.

**Comment by stianjensen** [comment]
> That's what I was trying to pull off on Friday, to no avail. Can you show how that would work? If not I'll go with Object.assign.

Did you have an updated node version then?

**Comment by sigvef** [comment]
Nope, but I tried the same tricks after upgrading.

**Comment by stianjensen** [comment]
Use Object.assign, object spread syntax is still only a stage 3 proposal it turns out

**Comment by sigvef** [comment]
ping

**Pull request merged**